src/mmw/js/src/analyze/models.js

```diff
@@ -141,89 +141,210 @@ var AnalyzeTaskCollection = Backbone.Collection.extend({
+                    area_of_interest: aoi,
+                    wkaoi: wkaoi,
+                    taskName: "analyze/soil",
+                    enabledForCatalogMode: true,
+                },
+            ]
         },
         {
             name: "terrain",
             displayName: "Terrain",
-            area_of_interest: aoi,
-            wkaoi: wkaoi,
-            taskName: "analyze/terrain"
+            tasks: [
+                {
+                    name: "terrain",
+                    area_of_interest: aoi,
+                    wkaoi: wkaoi,
+                    taskName: "analyze/terrain"
+                }
+            ]
         },
         {
             name: "climate",
             displayName: "Climate",
-            area_of_interest: aoi,
-            wkaoi: wkaoi,
-            taskName: "analyze/climate",
-            enabledForCatalogMode: true,
+            tasks: [
+                {
+                    name: "climate",
+                    area_of_interest: aoi,
+                    wkaoi: wkaoi,
+                    taskName: "analyze/climate",
+                    enabledForCatalogMode: true,
+                },
+            ]
         },
         {
             name: "pointsource",
             displayName: "Pt Sources",
-            area_of_interest: aoi,
-            wkaoi: wkaoi,
-            taskName: "analyze/pointsource"
+            tasks: [
+                {
+                    name: "pointsource",
+                    area_of_interest: aoi,
+                    wkaoi: wkaoi,
+                    taskName: "analyze/pointsource"
+                }
+            ]
         },
         {
             name: "animals",
             displayName: "Animals",
-            area_of_interest: aoi,
-            wkaoi: wkaoi,
-            taskName: "analyze/animals"
+            tasks: [
+                {
+                    name: "animals",
+                    area_of_interest: aoi,
+                    wkaoi: wkaoi,
+                    taskName: "analyze/animals"
+                }
+            ]
         },
         {
             name: "catchment_water_quality",
             displayName: "Water Qual",
-            area_of_interest: aoi,
-            wkaoi: wkaoi,
-            taskName: "analyze/catchment-water-quality"
-        },
-        {
-            name: "protected_lands",
-            displayName: "Protected Lands",
-            area_of_interest: aoi,
-            wkaoi: wkaoi,
-            taskName: "analyze/protected-lands"
+            tasks: [
+                {
+                    name: "catchment_water_quality",
+                    area_of_interest: aoi,
+                    wkaoi: wkaoi,
+                    taskName: "analyze/catchment-water-quality"
+                }
+            ]
         },
     ];
 
     // Remove analyses which aren't available for catalog search mode
     if (settings.get('data_catalog_enabled')) {
-        tasks = _.filter(tasks, { enabledForCatalogMode: true });
```

Comment:
This is a little easier to follow for me, how about you?

```diff
diff --git a/src/mmw/js/src/analyze/models.js b/src/mmw/js/src/analyze/models.js
index 43a32e5f..a1b7252c 100644
--- a/src/mmw/js/src/analyze/models.js
+++ b/src/mmw/js/src/analyze/models.js
@@ -318,18 +318,18 @@ function createAnalyzeTaskGroupCollection(aoi, wkaoi) {
         },
     ];
 
-    // Remove analyses which aren't available for catalog search mode
     if (settings.get('data_catalog_enabled')) {
-        var filteredTaskGroups = [];
-        _.foreach(taskGroups, function(tg) {
-            var filteredTasks = _.filter(tg.tasks, { enabledForCatalogMode: true });
-            if(filteredTasks.length > 0) {
-                tg.tasks = filteredTasks;
-                filteredTaskGroups.push(tg);
-            }
-        });
-
-        taskGroups = filteredTaskGroups;
+        taskGroups = _(taskGroups)
+            // Remove tasks not supported in data catalog mode
+            .map(function(tg) {
+                tg.tasks = _.filter(tg.tasks, { enabledForCatalogMode: true });
+                return tg;
+            })
+            // Remove task groups with no tasks
+            .filter(function(tg) {
+                return !_.isEmpty(tg.tasks);
+            })
+            .value();
     }
 
     taskGroups = _.map(taskGroups, function (tg) {
```

Reply:
Reads better to me - the optim(ize)ist in me feels bad about the double iteration, but readability trumps 5 more loops for sure :)